File: iris-crypto/iris-crypto-rsa/src/main/java/city/smartb/iris/crypto/rsa/RSAKeyPairEncoderBase64.java
package city.smartb.iris.crypto.rsa;

import java.security.interfaces.RSAPrivateKey;
import java.security.interfaces.RSAPublicKey;
import java.util.Base64;

public class RSAKeyPairEncoderBase64 {

    public static String encodePrivateKey(RSAPrivateKey value) {
        return encode(value.getEncoded());
    }

    public static String encodePublicKey(RSAPublicKey value) {
        return encode(value.getEncoded());
    }

    private static String encode(byte[] value) {
        try {
            return Base64.getUrlEncoder().encodeToString(value);
        } catch (Exception e) {
            return Base64.getEncoder().encodeToString(value);
        }
    }


}

File: iris-crypto/iris-crypto-key-rsa/src/main/kotlin/city/smartb/iris/crypto/rsa/RSAKeyPairEncoderBase64.kt
package city.smartb.iris.crypto.rsa

import java.security.interfaces.RSAPrivateKey
import java.security.interfaces.RSAPublicKey
import java.util.Base64

object RSAKeyPairEncoderBase64 {

    fun encodePrivateKey(value: RSAPrivateKey): String {
        return encode(value.encoded)
    }

    fun encodePublicKey(value: RSAPublicKey): String {
        return encode(value.encoded)
    }

    private fun encode(value: ByteArray): String {
        return try {
            Base64.getUrlEncoder().encodeToString(value)
        } catch (e: Exception) {
            Base64.getEncoder().encodeToString(value)
        }
    }
}
